Find a functionally equivalent code:com.android.internal.policy.impl.PhoneWindow.preparePanel:COMMENT
Method Modifier: public      final       internal    
Comment:/**
 * Prepares the panel to either be opened or chorded. This creates the Menu
 * instance for the panel and populates it via the Activity callbacks.
 *
 * @param st The panel state to prepare.
 * @param event The event that triggered the preparing of the panel.
 * @return Whether the panel was prepared. If the panel should not be shown,
 * returns false.
 */

@@ -15,24 +15,24 @@
         st.createdPanelView = cb.onCreatePanelView(st.featureId);
     }
     final boolean isActionBarMenu = (st.featureId == FEATURE_OPTIONS_PANEL || st.featureId == FEATURE_ACTION_BAR);
-    if (isActionBarMenu && mActionBar != null) {
+    if (isActionBarMenu && mDecorContentParent != null) {
         // Enforce ordering guarantees around events so that the action bar never
         // dispatches menu-related events before the panel is prepared.
-        mActionBar.setMenuPrepared();
+        mDecorContentParent.setMenuPrepared();
     }
     if (st.createdPanelView == null) {
         // Init the panel state's menu--return false if init failed
@@ -28,38 +28,38 @@
                     return false;
                 }
             }
-            if (isActionBarMenu && mActionBar != null) {
+            if (isActionBarMenu && mDecorContentParent != null) {
                 if (mActionMenuPresenterCallback == null) {
                     mActionMenuPresenterCallback = new ActionMenuPresenterCallback();
                 }
-                mActionBar.setMenu(st.menu, mActionMenuPresenterCallback);
+                mDecorContentParent.setMenu(st.menu, mActionMenuPresenterCallback);
             }
             // Call callback, and return if it doesn't want to display menu.
             // Creating the panel menu will involve a lot of manipulation;
@@ -41,49 +41,49 @@
             if ((cb == null) || !cb.onCreatePanelMenu(st.featureId, st.menu)) {
                 // Ditch the menu created above
                 st.setMenu(null);
-                if (isActionBarMenu && mActionBar != null) {
+                if (isActionBarMenu && mDecorContentParent != null) {
                     // Don't show it in the action bar either
-                    mActionBar.setMenu(null, mActionMenuPresenterCallback);
+                    mDecorContentParent.setMenu(null, mActionMenuPresenterCallback);
                 }
                 return false;
             }
@@ -59,68 +59,68 @@
             st.frozenActionViewState = null;
         }
         if (!cb.onPreparePanel(st.featureId, st.createdPanelView, st.menu)) {
-            if (isActionBarMenu && mActionBar != null) {
+            if (isActionBarMenu && mDecorContentParent != null) {
                 // The app didn't want to show the menu for now but it still exists.
                 // Clear it out of the action bar.
-                mActionBar.setMenu(null, mActionMenuPresenterCallback);
+                mDecorContentParent.setMenu(null, mActionMenuPresenterCallback);
             }
             st.menu.startDispatchingItemsChanged();
             return false;

